crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -138,6 +138,16 @@ pub trait Converter: Send + Sync {
         }
     }
 
+    fn get_software_list_message(&mut self) -> Message {
+        match create_get_software_list_message() {
+            Ok(messages) => messages,
+            Err(error) => {
```

Comment:
If this function is called from the `try_convert` function as suggested in the previous comment, you don't even need to do this error mapping, as `try_convert` is allowed to throw errors.